src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -683,25 +671,16 @@ public int Seek(byte byte0, byte byte1, byte byte2)
                         {
                             var data = new Vector<byte>(array, index);
 
-                            var byte0Equals = Vector.Equals(data, GetVector(byte0));
-                            if (!byte0Equals.Equals(Vector<byte>.Zero))
-                            {
-                                byte0Index = LocateFirstFoundByte(ref byte0Equals);
-                            }
+                            var byteEquals = Vector.Equals(data, GetVector(byte0));
+                            byteEquals = Vector.ConditionalSelect(byteEquals, byteEquals, Vector.Equals(data, GetVector(byte1)));
+                            byteEquals = Vector.ConditionalSelect(byteEquals, byteEquals, Vector.Equals(data, GetVector(byte2)));
 
-                            var byte1Equals = Vector.Equals(data, GetVector(byte1));
-                            if (!byte1Equals.Equals(Vector<byte>.Zero))
+                            if (!byteEquals.Equals(Vector<byte>.Zero))
                             {
-                                byte1Index = LocateFirstFoundByte(ref byte1Equals);
+                                byteIndex = LocateFirstFoundByte(ref byteEquals);
```

Comment:
Please note that passing byteEquals by-ref would make it addr-exposed and would not be enregistered.  It would be better to make a copy of it and passing copy by ref.

e.g.
```
var temp = byteEquals;
byteIndex = LocateFirstFoundByte(temp);
```